be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -147,7 +155,15 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
     auto* runtime_state = _fragment_ctx->runtime_state();
     int func_version = request.__isset.func_version ? request.func_version : 2;
     runtime_state->set_func_version(func_version);
-    runtime_state->init_mem_trackers(query_id, wg != nullptr ? wg->mem_tracker() : nullptr);
+
+    if (wg && wg->is_check_big_query()) {
+        int64_t limit = ExecEnv::GetInstance()->query_pool_mem_tracker()->limit() * wg->get_mem_limit() *
```

Comment:
each WorkGroup already has a MemTracker whose limit is  query_pool_mem_tracker's limit  *  WorkGroup.mem_limit, so we just use wg->mem_tracker()->limit() * wg->get_big_query_limit() is ok